src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -66,30 +75,113 @@ public async Task<SecretResponseModel> GetAsync([FromRoute] Guid id)
         {
             throw new NotFoundException();
         }
+
+        if (!await userHasReadAccessToProject(secret))
+        {
+            throw new NotFoundException();
+        }
+
         return new SecretResponseModel(secret);
     }
 
-    [HttpGet("projects/{projectId}/secrets")]
-    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId)
+    [HttpGet("projects/{projectId}/{organizationId}/secrets")]
+    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId, [FromRoute] Guid organizationId)
     {
-        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId);
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId, userId, accessClient);
+
```

Comment:
Need to add the access secrets manager flag check here.
Instead of requiring organization ID in the URL, could do something like this:

```
  [HttpGet("projects/{projectId}/secrets")]
    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId)
    {
        var project = await _projectRepository.GetByIdAsync(projectId);
        if (project == null || !_currentContext.AccessSecretsManager(project.OrganizationId))
        {
            throw new NotFoundException();
        }
        
        var userId = _userService.GetProperUserId(User).Value;
        var orgAdmin = await _currentContext.OrganizationAdmin(project.OrganizationId);
        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);

        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId, userId, accessClient);
        return new SecretWithProjectsListResponseModel(secrets);
    }
```